### ISSUE TITLE
NUnit.Engine nuget package improvements

### ISSUE THREAD

**Issue by CharliePoole** [nunit/nunit#1297]
The nuget package for the engine currently puts the engine into the lib directory of the package. This causes all the engine assemblies to be referenced by the project to which it is added and all of the assemblies are copied to the output directory by default. This copying is the correct behavior for any application that uses the engine and wants to keep it in it's own directory, The references may or may not be needed:
 * nunit.engine.api - should always be referenced
 * nunit.engine - some apps may reference it directly if they intend to create it using new rather than through TestEngineActivator
 * nunit-agent and nunit-agent-x86 should not be referenced at all, but they must me in the same directory as the engine in order to be used.
 * Mono.Cecil assembly should be in the same directory as the engine but would not be referenced unless the application had a specific use for it.

Currently, it's necessary to manually remove the unwanted references. At a minimum we should automate this in an install.ps1 script.

However, other apps may not want to keep the engine local but may want to use a pre-existing engine installation. For those apps, we need an NUnit.Engine.Api package, which only installs the nunit.engine.api assembly and allows it to find the engine.

Finally, when developing runners such as the NUnit gui, it may be desirable to have a special version of the engine that is used (because new features are added in parallel) while developing but still rely on a separately installed engine at run time. This may lead to an engine package that keeps the engine in the tools directory, so that nothing is referenced or copied.

@rprouse I'm laying this out so we can decide how much to do for 3.2 and how much to save for the future. What are your thoughts?

**Comment by CharliePoole** [comment]
Whatever we leave out of 3.2 can go into a separate issue for the backlog.

**Comment by CharliePoole** [comment]
In a conversation between @rprouse and myself, we decided to stick with the engine in the lib directory as it is. There will be a separte experimental package that puts it in the tools directory, but we won't publish it on nuget.
